pandas/io/formats/info.py

```diff
@@ -72,92 +70,114 @@ def _sizeof_fmt(num: Union[int, float], size_qualifier: str) -> str:
     return f"{num:3.1f}{size_qualifier} PB"
 
 
-class BaseInfo(metaclass=ABCMeta):
+class BaseInfo(ABC):
+    """Base class for DataFrameInfo and SeriesInfo.
+
+    Parameters
+    ----------
+    data : FrameOrSeries
+        Either dataframe or series.
+    memory_usage : bool or str, optional
+        If "deep", introspect the data deeply by interrogating object dtypes
+        for system-level memory consumption, and include it in the returned
+        values.
+    """
+
     def __init__(
         self,
         data: FrameOrSeries,
-        verbose: Optional[bool] = None,
-        buf: Optional[IO[str]] = None,
-        max_cols: Optional[int] = None,
         memory_usage: Optional[Union[bool, str]] = None,
-        null_counts: Optional[bool] = None,
     ):
-        if buf is None:  # pragma: no cover
-            buf = sys.stdout
+        self.data = data
+        self.memory_usage = self._initialize_memory_usage(memory_usage)
+
+    def _initialize_memory_usage(
+        self,
```

Comment:
I don't think `self` is used here, can this be a static method or a module-level function?

Reply:
I made it a static method.